src/sidebar/oauth-auth.js

```diff
@@ -79,14 +80,20 @@ function auth($http, settings) {
 
   // Set a timeout to refresh the access token a few minutes before it expires.
   function refreshAccessTokenBeforeItExpires(tokenInfo) {
-    var delay = tokenInfo.expiresIn * 1000;
-
     // We actually have to refresh the access token _before_ it expires.
     // If the access token expires in one hour, this should refresh it in
     // about 55 mins.
-    delay = Math.floor(delay * 0.91);
+    var refreshAfter = tokenInfo.expiresAt.valueOf() * 0.91;
```

Comment:
So this value is getting re-computed every 5 seconds (always giving the same answer, until the access token changes). The wasted CPU cycles could be avoided (for example by putting `refreshAfter` in the `tokenInfo` object) but I went for simplicity (keeping the variable near the one place that uses it) instead.